annnufan/assignment4/islands_counter.cpp

```diff
@@ -0,0 +1,59 @@
+#include <vector>
+#include <utility>
+
+class islands_counter {
+	std::vector<std::vector<bool>> island_map;
+	std::vector<std::pair<int, int>> moves;
+	int n, m;
+	int islands;
+	bool is_count;

```

Comment:
Naming tips:

A reader may confuse the variable `islands` with `islands_map`. I think `num_islands` or `island_count` can make it clear and distinguishable and meaningful.

`is_count` is not intuitive to understand unless you see where it is used.
You can replace it with `already_count` or even `num_islands_already_calculated` or `has_result`.
Think of it as a part of an English sentence. 
`if (has_result)` or `if (already_count)` can read as a sentence and are clear for the reader.
Even long names are OK if they help readability.